pkg/tests/dbaas_operator_tests.go

```diff
@@ -54,6 +54,18 @@ var _ = Describe("Rhoda e2e Test", func() {
 			//loop through providers
 			for i := range providers {
 				provider := providers[i]
+				providerType := string(provider.SecretData["providerType"])
+				BeforeEach(func() {
```

Comment:
Given that each loop has a single (It) subject node, I assume the BeforeEach will run just once in each iteration? What is the impact of using BeforeEach in this case, versus doing the check in each subject node?

Reply:
Here BeforeEach is executed for the "It" once for each provider. If "It" were replaced with "DescribeTable" which has multiple test cases, then BeforeEach would run once for each test case, for instance:
https://github.com/RHEcosystemAppEng/dbaas-operator/blob/main/api/v1alpha1/dbaasconnection_webhook_test.go#L49

Reply:
That doesn't work, and afaiu, Context() is irrelevant. It() blocks define test specs and BeforeEach means run once for each of them. So again, I'm looking for any good reason to use BeforeEach() here. If it was "Before()", which unfortunately does not exist, then it would make sense. But I don't think this does.

Reply:
It will be something like below - note that multiple "It"s are inside a new Context().
				BeforeEach(func() {
					By(fmt.Sprintf("Checking if provider operator is ready: %s", provider.ProviderName))
					Eventually(func() bool {
						if err := client.Get(context.Background(), k8sClient.ObjectKey{
							Name: providerType,
						}, &dbaasv1alpha1.DBaaSProvider{}); err != nil {
							return false
						}
						return true
					}, time.Second*120, time.Second).Should(BeTrue())
				})
				Context("Test provider cases", func() {
					It("Should pass when secret and provider created and connection status is checked for "+provider.ProviderName, func() { ....})
					It("Should test case B pass, func() { ....})
					It("Should test case C pass, func() { ....})
                                  })